plugins/modules/nd_federation_member.py

```diff
@@ -0,0 +1,280 @@
+        else:
+            # Remove exisiting members not specified by the users.
+            member_exists = False
+            for existing_member_hosts in federation_member_obj:
+                if not existing_member_hosts.get("spec").get("host") == local_cluster_name:
+                    for user_member_host in clusters:
+                        if existing_member_hosts.get("spec").get("host") == user_member_host.get("cluster_hostname"):
+                            member_exists = True
+                            break
+                    if not member_exists:
+                        remove_member_list.append(existing_member_hosts)
+
+            # Add members specified by the users.
+            for user_member_host in clusters:
+                found = False
+                for existing_member_hosts in federation_member_obj:
+                    if existing_member_hosts.get("spec").get("host") == user_member_host.get("cluster_hostname"):
+                        found = True
+                        break
+                if not found:
+                    add_member_list.append(user_member_host)
+
+        if add_member_list or remove_member_list:
+            if remove_member_list:
+                for member in remove_member_list:
+                    cluster_member_path = "{0}/{1}".format(member_path, member.get("status").get("memberID"))
+                    nd.request(cluster_member_path, method="DELETE")
+
+                    try:
+                        payload_dict["DELETE"].append(cluster_member_path)
+                    except:
+                        payload_dict["DELETE"] = [cluster_member_path]
+
+            if add_member_list:
+                for member in add_member_list:
+                    cluster_payload = dict(
+                        spec=dict(
+                            host=member.get("cluster_hostname"),
+                            userName=member.get("cluster_username"),
+                            password=(base64.b64encode(str.encode(member.get("cluster_password")))).decode("utf-8"),
+                            loginDomain=member.get("cluster_login_domain"),
+                        ),
+                    )
+
+                    payload = cluster_payload
+
+                    try:
+                        payload_dict["POST"].append(payload)
+                    except:
+                        payload_dict["POST"] = [payload]
+
+                    nd.sanitize(payload, collate=True)
+
+                    if not module.check_mode:
+                        # If federation does not exist, create a new federation
+                        if not federation_obj:
+                            nd.request(federation_path, method="POST", data={"spec": {"name": local_cluster_name}})
+                        nd.request(member_path, method="POST", data=payload)
+
+            if not module.check_mode:
+                nd.existing = nd.query_obj(member_path, ignore_not_found_error=True).get("items")
```

Comment:
why do we do another query here?

Reply:
I am trying to send a list of members in the payload, which is not allowed by the API, so i have to make one API call for each member in the list -> due to which  nd.existing is usually the last list member sent. which causes issue for idempotency. so, making a query here